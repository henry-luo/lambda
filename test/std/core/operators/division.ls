// Test: Division Operations
// Category: core/operators
// Type: positive
// Expected: 4

// Test division with variables
let x = 20
let y = 5
x / y
