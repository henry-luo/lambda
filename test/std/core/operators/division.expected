09:15:47 [NOTICE] Start transpiling test/std/core/operators/division.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:x:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:y:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr /.7:float]
      [primary expr:int,const:1]
        [ident:x:int,const:1]
      [primary expr:int,const:1]
        [ident:y:int,const:1]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/division.ls' to MIR
C2MIR init end           -- 63 usec
test/std/core/operators/division.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2365, parse tokens -- 1415
  C2MIR preprocessor end    -- 453 usec
  C2MIR parser end          -- 591 usec
  C2MIR context checker end -- 820 usec
  C2MIR generator end       -- 896 usec
C2MIR compiler end                -- 932 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/division.ls' executed: #####
4
